import.sh

```diff
@@ -2,195 +2,29 @@
-src/google/protobuf/compiler/java/java_message.cc
-src/google/protobuf/compiler/java/java_message_builder.cc
-src/google/protobuf/compiler/java/java_message_builder_lite.cc
-src/google/protobuf/compiler/java/java_message_field.cc
-src/google/protobuf/compiler/java/java_message_field_lite.cc
-src/google/protobuf/compiler/java/java_message_lite.cc
-src/google/protobuf/compiler/java/java_name_resolver.cc
-src/google/protobuf/compiler/java/java_primitive_field.cc
-src/google/protobuf/compiler/java/java_primitive_field_lite.cc
-src/google/protobuf/compiler/java/java_service.cc
-src/google/protobuf/compiler/java/java_shared_code_generator.cc
-src/google/protobuf/compiler/java/java_string_field.cc
-src/google/protobuf/compiler/java/java_string_field_lite.cc
-src/google/protobuf/compiler/javanano/javanano_enum.cc
-src/google/protobuf/compiler/javanano/javanano_enum_field.cc
-src/google/protobuf/compiler/javanano/javanano_extension.cc
-src/google/protobuf/compiler/javanano/javanano_field.cc
-src/google/protobuf/compiler/javanano/javanano_file.cc
-src/google/protobuf/compiler/javanano/javanano_generator.cc
-src/google/protobuf/compiler/javanano/javanano_helpers.cc
-src/google/protobuf/compiler/javanano/javanano_map_field.cc
-src/google/protobuf/compiler/javanano/javanano_message.cc
-src/google/protobuf/compiler/javanano/javanano_message_field.cc
-src/google/protobuf/compiler/javanano/javanano_primitive_field.cc
-src/google/protobuf/compiler/js/js_generator.cc
-src/google/protobuf/compiler/objectivec/objectivec_enum.cc
-src/google/protobuf/compiler/objectivec/objectivec_enum_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_extension.cc
-src/google/protobuf/compiler/objectivec/objectivec_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_file.cc
-src/google/protobuf/compiler/objectivec/objectivec_generator.cc
-src/google/protobuf/compiler/objectivec/objectivec_helpers.cc
-src/google/protobuf/compiler/objectivec/objectivec_map_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_message.cc
-src/google/protobuf/compiler/objectivec/objectivec_message_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_oneof.cc
-src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc
-src/google/protobuf/compiler/plugin.cc
-src/google/protobuf/compiler/plugin.pb.cc
-src/google/protobuf/compiler/python/python_generator.cc
-src/google/protobuf/compiler/ruby/ruby_generator.cc
-src/google/protobuf/compiler/subprocess.cc
-src/google/protobuf/compiler/zip_writer.cc
-'
+# symlink so cgo compiles them
+for file in $LIBPROTOBUF_LITE_SOURCES $LIBPROTOBUF_SOURCES; do
+  ln -sf internal/src/$file .
+done
 
 # symlink so cgo compiles them
-for file in $PROTOC_SOURCES; do
-  ln -sf ../internal/$file libprotoc
+for file in $LIBPROTOC_SOURCES; do
+  ln -sf ../../internal/src/$file cmd/protoc
 done
+
+# restore the repo to what it would look like when first cloned.
+# comment this line out while updating upstream.
+git clean -dxf
```

Comment:
Isn't updating upstream the common case for running `import.sh`? When would you run this other than when you're updating upstream?